tkg/clusterclient/clusterclient.go

```diff
@@ -1206,19 +1211,61 @@ func (c *client) GetBomConfigMap(tkrNameLabel string) (corev1.ConfigMap, error)
 	return cmList.Items[0], nil
 }
 
-// GetClusterInfrastructure gets the underlying infrastructure being used
-func (c *client) GetClusterInfrastructure() (string, error) {
+// getManagementCluster searches the cluster list for the one with management cluster role label
+func (c *client) getManagementCluster() (*capi.Cluster, error) {
 	clusters := &capi.ClusterList{}
 
 	selectors := []crtclient.ListOption{
 		crtclient.MatchingLabels(map[string]string{tkrconstants.ManagementClusterRoleLabel: ""}),
 	}
 	err := c.clientSet.List(context.Background(), clusters, selectors...)
 	if err != nil || len(clusters.Items) != 1 {
-		return "", errors.Wrap(err, "unable to get current management cluster")
+		return nil, errors.Wrap(err, "unable to get current management cluster")
+	}
+	return &clusters.Items[0], nil
+}
+
+// GetClusterInfrastructure gets the underlying infrastructure being used
+func (c *client) GetClusterInfrastructure() (string, error) {
+	cluster, err := c.getManagementCluster()
+	if err != nil {
+		return "", err
 	}
+	return cluster.Spec.InfrastructureRef.Kind, nil
+}
 
-	return clusters.Items[0].Spec.InfrastructureRef.Kind, nil
+// GetClusterResolvedTanzuKubernetesRelease gets the resolved TKR for the management cluster
+func (c *client) GetClusterResolvedTanzuKubernetesRelease() (*v1alpha3.TanzuKubernetesRelease, error) {
+	cluster, err := c.getManagementCluster()
+	if err != nil {
+		return nil, err
+	}
+	tkrName, exists := cluster.Labels[runv1alpha3.LabelTKR]
+	if !exists {
+		return nil, nil // the cluster doesn't have resolved TKR
+	}
+	var tkr v1alpha3.TanzuKubernetesRelease
+	if err := c.GetResource(&tkr, tkrName, "", nil, nil); err != nil {
+		return nil, err
+	}
+	return &tkr, nil
+}
+
+// GetClusterResolvedOSImages get a list of OSImage resource from the resolved TKR
+func (c *client) GetClusterResolvedOSImages() ([]runv1alpha3.OSImage, error) {
+	tkr, err := c.GetClusterResolvedTanzuKubernetesRelease()
+	if err != nil {
+		return []v1alpha3.OSImage{}, nil
+	}
```

Comment:
No need to call `c.GetClusterResolvedTanzuKubernetesRelease()`: just pass `tkr` as an argument. 

Also, return `[]*runv1alpha3.OSImage` instead of `[]runv1alpha3.OSImage`: to avoid unnecessary copying (e.g. when constructing the list) and possible bugs when iterating over the slice.

```suggestion
func (c *client) GetClusterResolvedOSImages(tkr *runv1alpha3.TanzuKubernetesRelease) ([]*runv1alpha3.OSImage, error) {
```